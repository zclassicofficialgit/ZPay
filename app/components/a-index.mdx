---
name: Home
route: /
---

# Zepio Wallet

Use the sidebar menu to browse the component library of React components available for the Zepio application.

> This is not an exhaustive list, and we're always looking for folks to help keep it updated.
