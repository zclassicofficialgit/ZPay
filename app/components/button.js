// @flow

import React, { type ElementProps } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const DefaultButton = styled.button`
  align-items: center;
  display: flex;
  justify-content: center;
  padding: 10px 30px;
  font-family: ${props => props.theme.fontFamily};
  font-weight: ${props => String(props.theme.fontWeight.bold)};
  font-size: ${props => `${props.theme.fontSize.regular}em`};
  cursor: pointer;
  outline: none;
  min-width: 100px;
  border-radius: ${props => props.theme.boxBorderRadius};
  transition: background-color 0.1s ${props => props.theme.transitionEase};
`;

const Primary = styled(DefaultButton)`
  background-color: ${props => props.theme.colors.buttonPrimaryBg};
  color: ${props => props.theme.colors.buttonPrimaryText};
  border: none;

  &:hover {
    opacity: 0.9;
  }

  &:disabled {
    background-color: ${props => props.theme.colors.buttonPrimaryDisabledBg};
    cursor: not-allowed;
    border: none;
    opacity: 0.45;

    &:hover {
      opacity: 0.45;
    }
  }
`;

const Secondary = styled(DefaultButton)`
  background-color: ${props => props.theme.colors.buttonSecondaryBg};
  color: ${props => props.theme.colors.buttonSecondaryText};
  border: 1px solid ${props => props.theme.colors.buttonSecondaryBorder};

  &:hover {
    background-color: ${props => props.theme.colors.buttonSecondaryHoveredBg}
    border-color: ${props => props.theme.colors.buttonSecondaryHoveredBg}
  }

  &:disabled {
    background-color: ${props => props.theme.colors.buttonSecondaryDisabledBg};
    cursor: not-allowed;
    color: ${props => props.theme.colors.buttonBorderColor};

    &:hover {
      border-color: ${props => props.theme.colors.buttonBorderColor};
    }
  }
`;

const Icon = styled.img`
  height: 12px;
  width: 12px;
  margin-right: 10px;
`;

type Props = {
  ...ElementProps<'button'>,
  label: string,
  onClick?: () => void,
  to?: ?string,
  variant?: 'primary' | 'secondary',
  disabled?: boolean,
  icon?: string | null,
  className?: string,
  isLoading?: boolean,
  id?: string,
};

export const Button = ({
  onClick,
  label,
  to,
  variant,
  disabled,
  icon,
  className,
  isLoading,
  id,
}: Props) => {
  if (to && onClick) throw new Error('Should define either "to" or "onClick"');

  const props = {
    onClick,
    disabled: disabled || isLoading,
    icon: null,
    className,
    id,
  };
  const buttonLabel = isLoading ? 'Loading...' : label;

  const component = variant === 'primary' ? (
    <Primary {...props} data-testid='PrimaryButton'>
      {icon ? <Icon src={icon} /> : null}
      {buttonLabel}
    </Primary>
  ) : (
    <Secondary {...props} data-testid='SecondaryButton'>
      {icon ? <Icon src={icon} /> : null}
      {buttonLabel}
    </Secondary>
  );

  return to ? <Link to={String(to)}>{component}</Link> : component;
};

Button.defaultProps = {
  to: '',
  icon: null,
  variant: 'primary',
  onClick: () => {},
  disabled: false,
  className: '',
  isLoading: false,
  id: '',
};
