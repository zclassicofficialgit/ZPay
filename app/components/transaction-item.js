// @flow

import React from 'react';
import styled, { withTheme } from 'styled-components';
import dateFns from 'date-fns';

import { DARK } from '../constants/themes';

import SentIconDark from '../assets/images/transaction_sent_icon_dark.svg';
import ReceivedIconDark from '../assets/images/transaction_received_icon_dark.svg';
import SentIconLight from '../assets/images/transaction_sent_icon_light.svg';
import ReceivedIconLight from '../assets/images/transaction_received_icon_light.svg';
import UnconfirmedLight from '../assets/images/unconfirmed_light.svg';
import UnconfirmedDark from '../assets/images/unconfirmed_dark.svg';

import { RowComponent } from './row';
import { ColumnComponent } from './column';
import { TextComponent } from './text';
import { ModalComponent } from './modal';
import { TransactionDetailsComponent } from './transaction-details';

import { formatNumber } from '../utils/format-number';
import { getCoinName } from '../utils/get-coin-name';

const Wrapper = styled(RowComponent)`
  background-color: ${props => props.theme.colors.transactionItemBg};
  padding: 15px 17px;
  cursor: pointer;
  border: 1px solid ${props => props.theme.colors.transactionItemBorder};
  border-bottom: none;

  &:last-child {
    border-bottom: 1px solid ${props => props.theme.colors.transactionItemBorder};
  }

  &:hover {
    background-color: ${props => props.theme.colors.transactionItemHoverBg};
  }
`;

const Icon = styled.img`
  width: 20px;
  height: 20px;
`;

/* eslint-disable max-len */
const TransactionTypeLabel = styled(TextComponent)`
  color: ${(props: PropsWithTheme<{ isReceived: boolean }>) => (props.isReceived ? props.theme.colors.transactionReceived : props.theme.colors.transactionSent)};
  text-transform: capitalize;
`;
/* eslint-enable max-len */

const TransactionAddress = styled(TextComponent)`
  color: ${props => props.theme.colors.transactionItemAddress};

  ${String(Wrapper)}:hover & {
    color: ${props => props.theme.colors.transactionItemAddressHover};
  }
`;

const TransactionLabel = styled(TextComponent)`
  color: ${props => props.theme.colors.transactionLabelText};

  ${String(Wrapper)}:hover & {
    color: ${props => props.theme.colors.transactionLabelTextHovered};
  }
`;

const TransactionColumn = styled(ColumnComponent)`
  margin-left: 10px;
  margin-right: 80px;
  min-width: 60px;
`;

const UnconfirmedStatusWrapper = styled.div`
  right: 30px;
  position: absolute;
`;

const UnconfirmedStatus = styled.img`
  width: 20px;
  height: 20px;
  opacity: 0.6;

  ${String(Wrapper)}:hover & {
    opacity: 1;
  }
`;

const RelativeRowComponent = styled(RowComponent)`
  position: relative;
`;

export type Transaction = {
  confirmed: boolean,
  confirmations: number,
  type: 'send' | 'receive',
  date: string,
  address: string,
  amount: number,
  zclPrice: number,
  transactionId: string,
  theme: AppTheme,
};

const Component = ({
  confirmed,
  confirmations,
  type,
  date,
  address,
  amount,
  zclPrice,
  transactionId,
  theme,
}: Transaction) => {
  const coinName = getCoinName();

  const isReceived = type === 'receive';
  const transactionTime = dateFns.format(new Date(date), 'HH:mm A');
  const transactionValueInZcl = formatNumber({
    value: amount,
    append: `${isReceived ? '+' : '-'}${coinName} `,
  });
  const transactionValueInUsd = formatNumber({
    value: amount * zclPrice,
    append: `${isReceived ? '+' : '-'}USD $`,
  });

  const receivedIcon = theme.mode === DARK ? ReceivedIconDark : ReceivedIconLight;
  const sentIcon = theme.mode === DARK ? SentIconDark : SentIconLight;
  const unconfirmedIcon = theme.mode === DARK ? UnconfirmedLight : UnconfirmedDark;

  // TODO: style the tooltip correctly (overlay issue)
  // const showUnconfirmed = !confirmed || confirmations < 1 || address === '(Shielded)';
  const showUnconfirmed = false;

  return (
    <ModalComponent
      renderTrigger={toggleVisibility => (
        <Wrapper
          id={`transaction-item-${transactionId}`}
          alignItems='center'
          justifyContent='space-between'
          onClick={toggleVisibility}
        >
          <RowComponent alignItems='center'>
            <RelativeRowComponent alignItems='center'>
              <Icon src={isReceived ? receivedIcon : sentIcon} alt='Transaction Type Icon' />
              <TransactionColumn>
                <TransactionTypeLabel isReceived={isReceived} value={type} isBold />
                <TransactionLabel value={transactionTime} isReceived={isReceived} />
              </TransactionColumn>
              {showUnconfirmed && (
                <UnconfirmedStatusWrapper>
                  <UnconfirmedStatus src={unconfirmedIcon} />
                </UnconfirmedStatusWrapper>
              )}
            </RelativeRowComponent>
            <TransactionAddress value={address} />
          </RowComponent>
          <ColumnComponent alignItems='flex-end'>
            <TextComponent
              isBold
              value={transactionValueInZcl}
              color={isReceived ? theme.colors.transactionReceived : theme.colors.transactionSent}
            />
            <TransactionLabel value={transactionValueInUsd} />
          </ColumnComponent>
        </Wrapper>
      )}
    >
      {toggleVisibility => (
        <TransactionDetailsComponent
          amount={amount}
          date={date}
          address={address}
          confirmed={confirmed}
          confirmations={confirmations}
          transactionId={transactionId}
          handleClose={toggleVisibility}
          type={type}
          zclPrice={zclPrice}
        />
      )}
    </ModalComponent>
  );
};

export const TransactionItemComponent = withTheme(Component);
