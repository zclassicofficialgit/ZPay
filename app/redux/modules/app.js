// @flow

import electronStore from '../../../config/electron-store';

import { ZCLASSIC_NETWORK, EMBEDDED_DAEMON } from '../../constants/zclassic-network';
import { NODE_SYNC_TYPES } from '../../constants/node-sync-types';

import type { Action } from '../../types/redux';

export type State = {|
  isErrorModalVisible: boolean,
  error: string | null,
  nodeSyncProgress: number,
  nodeSyncType: 'ready' | 'syncing' | 'error',
  zclassicNetwork: string,
  embeddedDaemon: boolean,
  isRefetching: boolean,
|};

// Actions
export const SHOW_ERROR_MODAL = 'SHOW_ERROR_MODAL';
export const HIDE_ERROR_MODAL = 'HIDE_ERROR_MODAL';
export const UPDATE_NODE_SYNC_STATUS = 'UPDATE_NODE_SYNC_STATUS';

export const showErrorModal = ({ error }: { error: string }) => ({
  type: SHOW_ERROR_MODAL,
  payload: {
    error,
  },
});

export const closeErrorModal = () => ({
  type: HIDE_ERROR_MODAL,
  payload: {},
});

export const updateNodeSyncStatus = ({
  nodeSyncProgress,
  nodeSyncType,
}: {
  nodeSyncProgress: number,
  nodeSyncType: $PropertyType<State, 'nodeSyncType'>,
}) => ({
  type: UPDATE_NODE_SYNC_STATUS,
  payload: {
    nodeSyncProgress,
    nodeSyncType,
  },
});

const initialState: State = {
  isErrorModalVisible: false,
  error: null,
  nodeSyncProgress: 0,
  nodeSyncType: NODE_SYNC_TYPES.SYNCING,
  zclassicNetwork: electronStore.get(ZCLASSIC_NETWORK),
  embeddedDaemon: electronStore.get(EMBEDDED_DAEMON),
  isRefetching: false,
};

// eslint-disable-next-line
export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case SHOW_ERROR_MODAL:
      return { ...state, isErrorModalVisible: true, error: action.payload.error };
    case HIDE_ERROR_MODAL:
      return { ...state, isErrorModalVisible: false, error: null };
    case UPDATE_NODE_SYNC_STATUS:
      return {
        ...state,
        nodeSyncProgress: action.payload.nodeSyncProgress,
        nodeSyncType: action.payload.nodeSyncType,
      };
    default:
      return state;
  }
};
