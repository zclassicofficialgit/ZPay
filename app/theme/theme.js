// @flow

import theme from 'styled-theming';

import { DARK, LIGHT } from '../constants/themes';
import { typography } from './typography';
import { DARK_COLORS, LIGHT_COLORS } from './colors';

// Building color object from theme keys
// All themes must have the same variables as DARK_COLORS
const colorKeys = Object.keys(DARK_COLORS);

const colors: Object = {};
colorKeys.forEach((key: string) => {
  colors[key] = theme('mode', {
    [LIGHT]: LIGHT_COLORS[key],
    [DARK]: DARK_COLORS[key],
  });
});

export const appTheme: AppTheme = {
  // General
  mode: DARK,

  // Typography
  ...typography,

  // Sizes & Spacing
  sidebarWidth: '180px',
  headerHeight: '60px',
  layoutPaddingLeft: '35px',
  layoutPaddingRight: '35px',
  layoutContentPaddingTop: '20px',
  boxBorderRadius: '3px',

  // Misc
  transitionEase: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)',

  // Colors
  colors: { ...colors },
};
