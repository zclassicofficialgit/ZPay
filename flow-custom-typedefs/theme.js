// @flow

import { ThemeSet } from 'styled-theming';

type Colors = {
  // General
  text: ThemeSet,
  success: ThemeSet,
  error: ThemeSet,
  background: ThemeSet,
  divider: ThemeSet,
  activeItem: ThemeSet,
  inactiveItem: ThemeSet,
  modalItemLabel: ThemeSet,

  // Header
  headerTitle: ThemeSet,

  // Select
  selectButtonShadow: ThemeSet,

  // Send View (Additional Options Section)
  sendAdditionalOptionsBg: ThemeSet,
  sendAdditionalOptionsBorder: ThemeSet,
  sendAdditionalInputBg: ThemeSet,
  sendAdditionalInputText: ThemeSet,

  // Dropdown
  dropdownBg: ThemeSet,
  dropdownHoveredBg: ThemeSet,
  dropdownBorder: ThemeSet,
  dropdownIconBorder: ThemeSet,
  dropdownOpenedIconBorder: ThemeSet,

  // Card
  cardBackgroundColor: ThemeSet,
  sendCardBg: ThemeSet,
  sendCardBorder: ThemeSet,

  // Buttons
  buttonPrimaryBg: ThemeSet,
  buttonPrimaryText: ThemeSet,
  buttonPrimaryDisabledBg: ThemeSet,
  buttonSecondaryBg: ThemeSet,
  buttonSecondaryText: ThemeSet,
  buttonSecondaryDisabledBg: ThemeSet,
  buttonSecondaryBorder: ThemeSet,
  buttonSecondaryHoveredBg: ThemeSet,
  buttonBorderColor: ThemeSet,

  // Transactions
  transactionSent: ThemeSet,
  transactionReceived: ThemeSet,
  transactionsDate: ThemeSet,
  transactionsItemHovered: ThemeSet,
  transactionItemBg: ThemeSet,
  transactionItemHoverBg: ThemeSet,
  transactionItemAddress: ThemeSet,
  transactionItemAddressHover: ThemeSet,
  transactionDetailsShadow: ThemeSet,
  transactionDetailsBg: ThemeSet,
  transactionDetailsRowHover: ThemeSet,
  transactionDetailsDivider: ThemeSet,
  transactionDetailsLabel: ThemeSet,
  transactionLabelText: ThemeSet,
  transactionLabelTextHovered: ThemeSet,

  // Status Pill
  statusPillBg: ThemeSet,
  statusPillLabel: ThemeSet,
  statusPillBorder: ThemeSet,

  // Sidebar
  sidebarBg: ThemeSet,
  sidebarItem: ThemeSet,
  sidebarItemActive: ThemeSet,
  sidebarActiveItemBorder:  ThemeSet,
  sidebarBorderRight: ThemeSet,
  sidebarItemHovered: ThemeSet,
  sidebarHoveredItemLabel: ThemeSet,
  sidebarLogoGradientBegin: ThemeSet,
  sidebarLogoGradientEnd: ThemeSet,
  sidebarItemHoveredBg: ThemeSet,

  // QR Code
  qrCodeWrapperBg: ThemeSet,
  qrCodeWrapperBorder: ThemeSet,

  // Forms
  inputBg: ThemeSet,
  inputBorder: ThemeSet,
  inputBorderActive: ThemeSet,

  // Wallet Summary
  walletSummaryBg: ThemeSet,
  walletSummaryBorder: ThemeSet,

  // Wallet Address
  walletAddressBg: ThemeSet,
  walletAddressBorder: ThemeSet,
  walletAddressInput: ThemeSet,
  walletAddressInputHovered: ThemeSet,
  walletAddressTooltip: ThemeSet,
  walletAddressTooltipBg: ThemeSet,

  // Console
  consoleBg: ThemeSet,
  consoleBorder: ThemeSet,

  // Settings
  settingsCardBg: ThemeSet,
  settingsLearnMore: ThemeSet,
  settingsLearnMoreHovered: ThemeSet,

  // Loading
  loadingScreenBg: ThemeSet,
  loadingScreenText: ThemeSet,
  loadingScreenProgress: ThemeSet,
};

type FontSize = {
  large: number,
  medium: number,
  regular: number,
  small: number,
};

type FontWeight = {
  bold: number,
  default: number,
  light: number,
};

type AppTheme = {
  // General
  mode: string,

  // Typography
  fontFamily: string,
  fontWeight: FontWeight,
  fontSize: FontSize,

  // Colors
  colors: Colors,

  // Spacing
  sidebarWidth: string,
  headerHeight: string,
  layoutPaddingLeft: string,
  layoutPaddingRight: string,
  layoutContentPaddingTop: string,
  boxBorderRadius: string,

  // Misc
  transitionEase: string,
};

declare type PropsWithTheme<T = {}> = {
  ...T,
  theme: AppTheme,
};
