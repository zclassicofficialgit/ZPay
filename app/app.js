// @flow

import React, { Component, Fragment } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';

import { configureStore, history } from './redux/create';
import { Router } from './router/container';
import { appTheme as theme, GlobalStyle } from './theme';
import electronStore from '../config/electron-store';
import { DARK, LIGHT, THEME_MODE } from './constants/themes';

const store = configureStore({});

type Props = {};
type State = {
  themeMode: string,
};

const getInitialTheme = () => {
  const themeInStore = String(electronStore.get(THEME_MODE));

  if (themeInStore === DARK || themeInStore === LIGHT) return themeInStore;
  return DARK;
};

export class App extends Component<Props, State> {
  state = {
    themeMode: getInitialTheme(),
  };

  componentDidMount() {
    electronStore.set(THEME_MODE, getInitialTheme());

    electronStore.onDidChange(THEME_MODE, newValue => this.setState({ themeMode: newValue }));
  }

  render() {
    const { themeMode } = this.state;

    return (
      <ThemeProvider theme={{ ...theme, mode: themeMode }}>
        <Fragment>
          <GlobalStyle />
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <Router />
            </ConnectedRouter>
          </Provider>
        </Fragment>
      </ThemeProvider>
    );
  }
}
