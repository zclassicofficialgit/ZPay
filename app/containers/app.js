// @flow

import { connect } from 'react-redux';

import { closeErrorModal } from '../redux/modules/app';
import { LayoutComponent } from '../components/layout';

import type { Dispatch } from '../types/redux';
import type { AppState } from '../types/app-state';

export type MapStateToProps = {|
  isErrorModalVisible: boolean,
  zclassicNetwork: string,
  error: string | null,
  embeddedDaemon: boolean,
|};

const mapStateToProps = ({ app }: AppState): MapStateToProps => ({
  isErrorModalVisible: app.isErrorModalVisible,
  error: app.error,
  zclassicNetwork: app.zclassicNetwork,
  embeddedDaemon: app.embeddedDaemon,
});

export type MapDispatchToProps = {|
  closeErrorModal: () => void,
|};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => ({
  closeErrorModal: () => dispatch(closeErrorModal()),
});

// $FlowFixMe
export const AppContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(LayoutComponent);
