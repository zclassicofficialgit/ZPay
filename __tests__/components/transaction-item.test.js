// @flow

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import dateFns from 'date-fns';
import 'jest-dom/extend-expect';

import { TransactionItemComponent } from '../../app/components/transaction-item';
import { appTheme } from '../../app/theme';

let originalDate;
const fixedDate = new Date('2018-02-28T09:39:59');

beforeAll(() => {
  // $FlowFixMe
  dateFns.format = jest.fn(() => '17:01 PM');

  originalDate = global.Date;
  global.Date = class extends Date {
    constructor() {
      super();

      return fixedDate;
    }
  };
});
afterAll(() => {
  global.Date = originalDate;
  // $FlowFixMe
  dateFns.format.mockRestore();
  cleanup();
});

describe('<TransactionItem />', () => {
  test('should render a transaction item correctly', () => {
    const { container } = render(
      <ThemeProvider theme={appTheme}>
        <TransactionItemComponent
          confirmed
          confirmations={10}
          type='send'
          address='123456789123456789123456789123456789'
          transactionId='a0s9dujo23j0'
          amount={0.8652}
          date={new Date().toString()}
          zclPrice={2.94}
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
