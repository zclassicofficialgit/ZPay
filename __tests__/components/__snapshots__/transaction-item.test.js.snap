// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionItem /> should render a transaction item correctly 1`] = `
<div>
  <div
    class="sc-jzJRlG bgSReS sc-bdVaJa bWRfen"
    id="transaction-item-a0s9dujo23j0"
  >
    <div
      class="sc-bdVaJa eFCRqo"
    >
      <div
        class="sc-ckVGcZ gzxdCf sc-bdVaJa eFCRqo"
      >
        <img
          alt="Transaction Type Icon"
          class="sc-cSHVUG fQARKX"
          src="[object Object]"
        />
        <div
          class="sc-kGXeez iHFhdz sc-bwzfXH iiePXZ"
          width=""
        >
          <p
            class="sc-kAzzGY gABhMI sc-htpNat ccIOzB"
          >
            send
          </p>
          <p
            class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
          >
            17:01 PM
          </p>
        </div>
      </div>
      <p
        class="sc-chPdSV dTXjho sc-htpNat pRhzD"
      >
        123456789123456789123456789123456789
      </p>
    </div>
    <div
      class="sc-bwzfXH jExuEg"
      width=""
    >
      <p
        class="sc-htpNat jAbxuE"
      >
        -TAZ 0.865
      </p>
      <p
        class="sc-kgoBCf goviHQ sc-htpNat pRhzD"
      >
        -USD $2.544
      </p>
    </div>
  </div>
</div>
`;
